### PR TITLE
Fix interface mismatch

### PR DESCRIPTION
**Pull request by andriy-stechak**

### CODE REVIEW

File: pkg/hasher/hasher.go [hunk 1]
```diff
@@ -8,14 +8,27 @@ import (
...
 	hash := sha512.New()
-	hash.Write([]byte(password))
+	_, err := hash.Write([]byte(password))
 
-	return fmt.Sprintf("%x", hash.Sum(nil))
+	if err != nil {
+		return passwordHash, err
+	}
+
+	passwordHash = fmt.Sprintf("%x", hash.Sum(nil))
```
Comment (yushni) [inline pkg/hasher/hasher.go]: `fmt.Sprintf `not really necessary - you can do simlply - `return string(hash.Sum(nil)), nil`

File: pkg/hasher/hasher.go [hunk 2]
```diff
@@ -8,14 +8,27 @@ import (
 )
 
 // HashPassword returns sha512 hash string for provided password
-func HashPassword(password string) string {
+func HashPassword(password string) (string, error) {
+	var passwordHash string
 	hash := sha512.New()
-	hash.Write([]byte(password))
+	_, err := hash.Write([]byte(password))
```
Comment (yushni) [inline pkg/hasher/hasher.go]: I'd suggest use short error check syntax in this way 
```
if _, err := hash.Write([]byte(password)); err != nil {
    return "", err
}
```

Comment (yushni) [inline pkg/hasher/hasher.go]: usually in go we do not leave empty space before the err check

**Review by yushni — APPROVED** [review]
Good job

File: pkg/hasher/hasher.go (hunk 1 above)
Comment (andriy-stechak) [reply pkg/hasher/hasher.go]: If I place type casting here tests are failing.
```
say@say-HP-ProBook-440-G6:~/go/src/lgc/pkg/hasher$ go test
--- FAIL: TestHashPassword (0.00s)
    hasher_test.go:8: HashPassword() = "\x9bq\xd2$\xbdb\xf3x]\x96\xd4j\xd3\xea=s1\x9b\xfb\u0089\f\xaa\xda\xe2\xdf\xf7%\x19g<\xa7##\xc3ٛ\xa5\xc1\x1d|z\xccn\x14\xb8\xc5\xda\fFcG\\.\\:\xde\xf4os\xbc\xde\xc0C", want "9b71d224bd62f3785d96d46ad3ea3d73319bfbc2890caadae2dff72519673ca72323c3d99ba5c11d7c7acc6e14b8c5da0c4663475c2e5c3adef46f73bcdec043"
--- FAIL: TestCheckPasswordHash (0.00s)
    hasher_test.go:15: CheckPasswordHash() = false, want true
FAIL
exit status 1
FAIL	github.com/andriystech/lgc/pkg/hasher	0.001s
```

**Pull request merged**